### PR TITLE
Add android architecture component navigator

### PR DESCRIPTION
**Pull request by lion4ik**
I've added `AacNavigator` which makes a friendship between Cicerone and Navigation Architecture Component from Jetpack :). 

So, it makes a possibility to work with Cicerone and see navigation graph using  Navigation Architecture Component. It helps to see the map of navigation. Once NAC becomes stable it will be useful to use it in production I hope.

### CODE REVIEW

File: sample/src/main/java/ru/terrakok/cicerone/sample/ui/aac/AacBlankFragment.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_aac_blank, container, false);
+        Toolbar toolbar = rootView.findViewById(R.id.toolbar);
+        toolbar.setTitle("Aac blank fragment");
+        Button btn_aac_first = rootView.findViewById(R.id.btn_aac_first);
+        btn_aac_first.setOnClickListener(new View.OnClickListener() {
```
Comment (terrakok) [inline sample/src/main/java/ru/terrakok/cicerone/sample/ui/aac/AacBlankFragment.java]: what is name?

File: aac_navigation/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            minifyEnabled false
+        }
+        debug {}
+    }
+
+}
+
+dependencies {
+    //Cicerone
+    implementation project(':library')
```
Comment (terrakok) [inline aac_navigation/build.gradle]: api

File: aac_navigation/src/main/java/ru/terrakok/cicerone/android/aac/AacNavigator.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            fragmentReplace(((Replace) command).getScreen());
+        } else if (command instanceof BackTo) {
+            fragmentBackTo(((BackTo) command).getScreen());
+        } else if (command instanceof Back) {
+            fragmentBack();
+        }
+    }
+
+    private void fragmentReplace(Screen screen) {
+        fragmentBack();
```
[the remark this answers is not in the document]
Comment (lion4ik) [reply aac_navigation/src/main/java/ru/terrakok/cicerone/android/aac/AacNavigator.java]: probably, you mean asynchronous? Ok, I'll keep in mind.

**Comment by ajans** [comment]
Why did you close this instead of a merge?
As I am using the Jetpack Navigation Component already, but am struggling to find an abstraction for it that enables me to keep a clean architecture (and enable shared element transitions with Jetpack Navigation), maybe this would have helped.
But without support, I won't be looking deeper into it because I wouldn't want to swap out my whole navigation and implement it from scratch with this library without getting the benefits from the jetpack-library.

**Pull request closed without merging**